pallets/file-system/src/utils.rs

```diff
@@ -837,6 +814,135 @@ where
         Ok((bsp_id, new_root))
     }
 
+    pub(crate) fn do_delete_file(
+        sender: T::AccountId,
+        bucket_id: ProviderIdFor<T>,
+        file_key: MerkleHash<T>,
+        location: FileLocation<T>,
+        fingerprint: Fingerprint<T>,
+        size: StorageData<T>,
+        maybe_inclusion_forest_proof: Option<ForestProof<T>>,
+    ) -> Result<ProviderIdFor<T>, DispatchError> {
+        // Compute the file key hash.
+        let computed_file_key = Self::compute_file_key(
+            sender.clone(),
+            bucket_id,
+            location.clone(),
+            size,
+            fingerprint,
+        );
+
+        // Check that the metadata corresponds to the expected file key.
+        ensure!(
+            file_key == computed_file_key,
+            Error::<T>::InvalidFileKeyMetadata
+        );
+
+        let msp_id = <T::Providers as ReadProvidersInterface>::get_msp_of_bucket(&bucket_id)
```

Comment:
I'd add a check for whether this bucket belongs to that MSP, and if the bucket owner is this user.

Reply:
In this function, we are fetching the bucket data from storage directly. We are only getting the MSP from that data to report it in the event and record it in `PendingFileDeletionRequests` storage for MSPs to track and respond to.

Reply:
I added the check for the latter.